### ISSUE TITLE
Move tlogserver logic to client (i.e. : remove tlogserver)

### ISSUE THREAD

**Issue by iwanbk** [zero-os/0-Disk#419]
I propose to remove tlogserver and moves the logic to client 

I'm  thinking of this because there are many changes since the original tlog server at https://github.com/g8os/tlog:

- In Go : the initial implementation was in C++ and Nim. Being in Go, make it possible to implement it entirely in nbd
- using 0-stor : 0-stor reduces some complexities in the tlog server code
- no dpdk : i never heard plan toward dpdk anymore, CMIIW. Even dpdk would be slower than direct func call or Go channel

The advantages to move the logic to client:
- faster, as there is no communication overhead between tlog client and server
- reduces point of failure
- simpler communication : golang channel  vs async TCP / dpdk. No need to handle for unordered, lost, and  corrupted logs.
- current server is only an entity that buffer the log and send it to 0-stor
- no need to create redundancy

In short: the logic and thus the code will be simpler, performance will be better. Simpler logic tend to more reliable 

Additional loads caused by this change
- capnp aggregation creation. notes that it is only the aggregation, the client itself already creates capnp block when sending the block to server
- aggregation processing : encrypt, compress, erasure encoding

I think the additional load can be solved by giving more resource (CPU, RAM) which can be taken from resource for tlog servers.

**Comment by zaibon** [comment]
interesting proposition indeed. I think it make sense if we consider then now the tlog server is really doing nothing more then receiving data from nbdserver and most of the logic belong the in the 0-stor client.
although I don't think we should focus on this at the moment, I'll move to milestone beta-1 see if we have time to there and it gives us more time to really understand the impact of such a decisions.

On more cons I see is that if we remove the tlog server, we can't then spread the load anymore. In the future when zero-os will be used in real distributed scheme. we could imagine to have the tlog server running on a completely different location then the nbdserver.

**Comment by iwanbk** [comment]
> On more cons I see is that if we remove the tlog server, we can't then spread the load anymore. In the future when zero-os will be used in real distributed scheme. we could imagine to have the tlog server running on a completely different location then the nbdserver.

This argument is enough to close this issue :)
As one of the motivation is : 

> one tlog server for one nbdserver setup.

**Comment by GlenDC** [comment]
There is no real reason why you would limit it to one tlogserver for one nbdserver forever. What I mean is that sure things might get more complex, but it should definitely be possible to have multiple tlogservers working concurrent, as all they do is process blocks and form aggregations. I know this will only be possible once problems X and Y are solved, but if I remember correct all problems ever mentioned with it, they all were solvable.
